lib/ipam/ipam_test.go

```diff
@@ -103,364 +77,323 @@ type testArgsClaimAff struct {
 	// assigned IP to be from the new ipPool that was created, this is to make sure the assigned IP
 	// doesn't come from the old affinedBlock even after the ipPool was deleted.
 	Describe("IPAM AutoAssign from the default pool then delete the pool and assign again", func() {
-		c := testutils.CreateCleanClient(config)
-		ic := setupIPAMClient(c, true)
-
-		host := "host-A"
+		hostA := "host-A"
+		hostB := "host-B"
 		pool1 := cnet.MustParseNetwork("10.0.0.0/24")
+		pool2 := cnet.MustParseNetwork("20.0.0.0/24")
 		var block cnet.IPNet
 
-		testutils.CreateNewIPPool(*c, "10.0.0.0/24", false, false, true)
+		Context("AutoAssign a single IP without specifying a pool", func() {
 
-		// Step-1: AutoAssign 1 IP without specifying a pool - expect the assigned IP is from pool1.
-		Context("AutoAssign 1 IP without specifying a pool", func() {
-			args := client.AutoAssignArgs{
-				Num4:     1,
-				Num6:     0,
-				Hostname: host,
-			}
+			It("should auto-assign from the only available pool", func() {
+				bc.Clean()
+				deleteAllPools()
+				applyPool("10.0.0.0/24", true)
 
-			v4, _, outErr := ic.AutoAssign(args)
+				args := AutoAssignArgs{
+					Num4:     1,
+					Num6:     0,
+					Hostname: hostA,
+				}
 
-			blocks := getAffineBlocks(host)
+				v4, _, outErr := ic.AutoAssign(args)
 
-			for _, b := range blocks {
-				if pool1.Contains(b.IPNet.IP) {
-					block = b
+				blocks := getAffineBlocks(bc, hostA)
+				for _, b := range blocks {
+					if pool1.Contains(b.IPNet.IP) {
+						block = b
+					}
 				}
-			}
-
-			It("assigned IP should be from pool1", func() {
 				Expect(outErr).NotTo(HaveOccurred())
 				Expect(pool1.IPNet.Contains(v4[0].IP)).To(BeTrue())
 			})
-		})
-
-		// Step-2: AutoAssign 1 more IP without specifying a pool - expect the assigned IP
-		// is from the same /26 block as the previous IP.
-		Context("AutoAssign 1 IP without specifying a pool", func() {
-			args := client.AutoAssignArgs{
-				Num4:     1,
-				Num6:     0,
-				Hostname: host,
-			}
 
-			v4, _, outErr := ic.AutoAssign(args)
+			It("should auto-assign another IP from the same pool into the same allocation block", func() {
+				args := AutoAssignArgs{
+					Num4:     1,
+					Num6:     0,
+					Hostname: hostA,
+				}
 
-			It("assigned IP should be from pool1", func() {
+				v4, _, outErr := ic.AutoAssign(args)
 				Expect(outErr).NotTo(HaveOccurred())
 				Expect(block.IPNet.Contains(v4[0].IP)).To(BeTrue())
 			})
-		})
 
-		// Step-3: Delete pool1 - expect it to execute without any error.
-		Context("Delete pool1", func() {
-			outErr := c.IPPools().Delete(api.IPPoolMetadata{
-				CIDR: pool1,
-			})
+			It("should assign from a new pool for a new host (old pool is removed)", func() {
+				deleteAllPools()
+				applyPool("20.0.0.0/24", true)
+
+				p, _ := ipPools.GetEnabledPools(4)
+				Expect(len(p)).To(Equal(1))
+				Expect(p[0].String()).To(Equal(pool2.String()))
+				p, _ = ipPools.GetEnabledPools(6)
+				Expect(len(p)).To(Equal(0))
```

Comment:
fyi: gomega also has a `BeZero()` function. Not saying you need to change this

Reply:
good to know.